viewer/src/public/RevealManager.ts

```diff
@@ -2,50 +2,225 @@
  * Copyright 2020 Cognite AS
  */
 
-import { RevealManagerBase } from './RevealManagerBase';
-import { CogniteClient } from '@cognite/sdk';
-import { CdfModelDataClient } from '@/utilities/networking/CdfModelDataClient';
-import { PotreeGroupWrapper, PotreeNodeWrapper } from '@/internal';
-import { CadNode, NodeAppearanceProvider } from '@/datamodels/cad';
-import { RevealOptions } from './types';
-import omit from 'lodash/omit';
-import { initMetrics } from '@/utilities/metrics';
-
-type CdfModelIdentifier = { modelId: number; revisionId: number };
-
-export class RevealManager extends RevealManagerBase<CdfModelIdentifier> {
-  constructor(client: CogniteClient, options: RevealOptions = {}) {
-    initMetrics(options.logMetrics !== false, client.project, {
-      moduleName: 'RevealManager',
-      methodName: 'constructor',
-      constructorOptions: omit(options, ['internal'])
-    });
-    const clientExt = new CdfModelDataClient(client);
-    super(clientExt, options);
+import * as THREE from 'three';
+
+import { CadManager } from '@/datamodels/cad/CadManager';
+import { PointCloudManager } from '@/datamodels/pointcloud/PointCloudManager';
+import {
+  SectorNodeIdToTreeIndexMapLoadedListener,
+  SectorNodeIdToTreeIndexMapLoadedEvent,
+  LoadingStateChangeListener
+} from './types';
+import { Subscription, combineLatest } from 'rxjs';
+import { distinctUntilChanged, map, share, filter } from 'rxjs/operators';
+import { trackError, trackLoadModel } from '@/utilities/metrics';
+import { NodeAppearanceProvider, CadNode } from '@/datamodels/cad';
+import { PotreeGroupWrapper } from '@/datamodels/pointcloud/PotreeGroupWrapper';
+import { PotreeNodeWrapper } from '@/datamodels/pointcloud/PotreeNodeWrapper';
+import { RenderMode } from '@/datamodels/cad/rendering/RenderMode';
+
+export class RevealManager<TModelIdentifier> {
+  private readonly _cadManager: CadManager<TModelIdentifier>;
+  private readonly _pointCloudManager: PointCloudManager<TModelIdentifier>;
+
+  private readonly _lastCamera = {
+    position: new THREE.Vector3(NaN, NaN, NaN),
+    quaternion: new THREE.Quaternion(NaN, NaN, NaN, NaN),
+    zoom: NaN
+  };
+
+  private _isDisposed = false;
+  private readonly _subscriptions = new Subscription();
+  private readonly eventListeners = {
+    sectorNodeIdToTreeIndexMapLoaded: new Array<SectorNodeIdToTreeIndexMapLoadedListener>(),
+    loadingStateChanged: new Array<LoadingStateChangeListener>()
+  };
+
+  constructor(cadManager: CadManager<TModelIdentifier>, pointCloudManager: PointCloudManager<TModelIdentifier>) {
```

Comment:
Probably should be marked with `/** @interhal */` if you don't expect any outside usage of it

Reply:
marked it with internal for now :)